src/conn/conn_api.c

```diff
@@ -1816,6 +1816,48 @@ __conn_single(WT_SESSION_IMPL *session, const char *cfg[])
 	return (ret);
 }
 
+/*
+ * __wt_diagnostic_config --
+ *	Set verbose configuration.
+ */
+int
+__wt_diagnostic_config(WT_SESSION_IMPL *session, const char *cfg[])
+{
+	WT_CONFIG_ITEM cval;
+	WT_CONNECTION_IMPL *conn;
+	WT_TXN_GLOBAL *txn_global;
+
+	conn = S2C(session);
+	txn_global = &conn->txn_global;
+
+	WT_RET(__wt_config_gets(session,
+	    cfg, "diagnostic.rollback_error", &cval));
+	txn_global->diag_rollback = (uint64_t)cval.val;
+
+	WT_RET(__wt_config_gets(session,
+	    cfg, "diagnostic.table_logging", &cval));
+	if (cval.val)
+		FLD_SET(conn->log_flags, WT_CONN_LOG_DIAGNOSTICS);
```

Comment:
This flag seems a bit generic to me - I can think of lots of things diagnostic log might enable - maybe this will become not relevant if you make a name change suggestion I made above. I'm going to hold off on further review for now.

Reply:
I will think on a more specific name for this flag and its effects.